configuration/pih/reports/sql/covid19encounterexport.sql

```diff
@@ -6,98 +6,1328 @@
-where
-    pr.voided = 0 and
-    p.voided = 0 and
-    e.voided = 0 and
-    et.name in ('COVID-19 Admission', 'COVID-19 Progress', 'COVID-19 Discharge');
-
-## REMOVE TEST PATIENTS
-
-delete
-from temp_encounter
-where patient_id in
-      (
-          select a.person_id
-          from person_attribute a
-                   inner join person_attribute_type t on a.person_attribute_type_id = t.person_attribute_type_id
-          where a.value = 'true'
-            and t.name = 'Test Patient'
-      );
-
-# ADD DETAILS FOR PATIENT
-
-update temp_encounter set dossier_num = dosId(patient_id);
-update temp_encounter set zlemr_id = zlemr(patient_id);
-update temp_encounter set address = person_address(patient_id);
-update temp_encounter set phone_number = person_attribute_value(patient_id, 'Telephone Number');
-
-# ADD DETAILS FOR ENCOUNTER
-
-update temp_encounter set encounter_provider = provider(encounter_id);
-update temp_encounter set encounter_location = encounter_location_name(encounter_id);
-
-# ADD OBSERVATIONS
-## TODO
-
-# EXECUTE SELECT TO EXPORT TABLE CONTENTS
-
-SELECT e.encounter_id,
-       e.patient_id,
-       e.dossier_num       as dossierId,
-       e.zlemr_id          as zlemr,
-       e.gender,
-       e.birthdate,
-       e.address,
-       e.phone_number,
-       e.encounter_type,
-       e.encounter_location,
-       e.encounter_datetime,
-       e.encounter_provider
-FROM temp_encounter e
-;
+        INNER JOIN patient p ON p.patient_id = e.patient_id
+        INNER JOIN person pr ON pr.person_id = e.patient_id
+        LEFT JOIN encounter_type et ON et.encounter_type_id = e.encounter_type
+WHERE
+    DATE(e.encounter_datetime) >=  @startDate AND
+    DATE(e.encounter_datetime) <=  @endDate AND
+    pr.voided = 0 AND
+    p.voided = 0 AND
+    e.voided = 0 AND
+    et.name IN ('COVID-19 Admission', 'COVID-19 Progress', 'COVID-19 Discharge');
+
+DELETE FROM temp_encounter 
+WHERE
+    patient_id IN (SELECT 
+        a.person_id
+    FROM
+        person_attribute a
+            INNER JOIN
+        person_attribute_type t ON a.person_attribute_type_id = t.person_attribute_type_id
+    
+    WHERE
+        a.value = 'true'
+        AND t.name = 'Test Patient');
+
+UPDATE temp_encounter 
+SET 
+    dossier_num = DOSID(patient_id);
+UPDATE temp_encounter 
+SET 
+    zlemr_id = ZLEMR(patient_id);
+UPDATE temp_encounter 
+SET 
+    address = PERSON_ADDRESS(patient_id);
+UPDATE temp_encounter 
+SET 
+    phone_number = PERSON_ATTRIBUTE_VALUE(patient_id, 'Telephone Number');
+
+UPDATE temp_encounter 
+SET 
+    encounter_provider = PROVIDER(encounter_id);
+UPDATE temp_encounter 
+SET 
+    encounter_location = ENCOUNTER_LOCATION_NAME(encounter_id);
+
+UPDATE temp_encounter 
+SET 
+    health_care_worker = OBS_VALUE_CODED_LIST(encounter_id, 'CIEL', '5619', 'fr');
+
+UPDATE temp_encounter 
+SET 
+    pregnant = OBS_VALUE_CODED_LIST(encounter_id, 'CIEL', '5272', 'fr');
+
+-- last menstruation date
+UPDATE temp_encounter te
```

Comment:
If we have to do a join here, we are missing a function.  Why are we doing this instead of adding the appropriate function to retrieve an obs value datetime?